### PR TITLE
bcm283x: Implement FastOut and corresponding benchmark

### PR DESCRIPTION
**Pull request by maruel**
Paging @simokawa .

I started writing a blog post about it at https://github.com/periph/website/blob/blog_gpio_perf/site/content/news/2017/gpio_perf.md

### CODE REVIEW

File: host/bcm283x/gpio.go [hunk 1]
```diff
@@ -266,6 +266,16 @@ func (p *Pin) Out(l gpio.Level) error {
 		return err
 	}
 	// Change output before changing mode to not create any glitch.
+	p.FastOut(l)
+	p.setFunction(out)
+	return nil
+}
+
+// FastOut sets a pin output level with Absolutely No error checking.
```
Comment (mattetti) [inline host/bcm283x/gpio.go]: We might want to specify what kind of errors could happen and why/when to use that version vs `Out`.

File: host/bcm283x/gpio.go [hunk 2]
```diff
@@ -266,6 +266,16 @@ func (p *Pin) Out(l gpio.Level) error {
...
+//
+// Out() Must be called once first, then FastOut() can be used for minimal CPU
```
Comment (mattetti) [inline host/bcm283x/gpio.go]: what happens of Out isn't called first?

Comment (maruel) [reply host/bcm283x/gpio.go]: Undefined behavior. Updated the comment to be clearer, wdyt?

File: host/bcm283x/gpio.go (hunk 1 above)
Comment (maruel) [reply host/bcm283x/gpio.go]: Updated comment to call out undefined behavior.

**Pull request merged**